Framework/NexusGeometry/src/NexusGeometryParser.cpp

```diff
@@ -672,5 +673,16 @@ NexusGeometryParser::createInstrument(const std::string &fileName) {
   auto rootGroup = file.openGroup("/");
   return extractInstrument(file, rootGroup);
 }
+
+// Create a unique instrument name from Nexus file
+std::string NexusGeometryParser::getMangledName(const std::string &fileName,
```

Comment:
I like this! Means that we can differentiate between different instruments sharing the same basic name.